packages/ra-core/src/dataProvider/index.ts

```diff
@@ -25,12 +24,9 @@ import useDeleteMany from './useDeleteMany';
 import useRefreshWhenVisible from './useRefreshWhenVisible';
 import useIsAutomaticRefreshEnabled from './useIsAutomaticRefreshEnabled';
 
-export type {
-    QueryOptions,
-    UseMutationValue,
-    UseQueryValue,
-    UseGetOneHookValue,
-};
+export * from './useQueryWithStore';
```

Comment:
This won't work: the `Query` type will conflict with the `Query` component.

The `Query` type name is too generic; Something like `DataProviderQuery` would be better, WDYT?